lib/fir/afforestation.h

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) 2018, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef FORTRAN_IR_AFFORESTATION_H_
+#define FORTRAN_IR_AFFORESTATION_H_
+
+namespace Fortran::parser {
+class Program;
+}
+
+namespace Fortran::semantics {
+class SemanticsContext;
+}
+
+namespace Fortran::IR {
```

Comment:
I think the namespace should be "fir" if the directory is.

Reply:
I disagree. It reads as doubly redundant.

Reply:
Then the directory and namespace should both be "ir". We shouldn't have two different names for the same thing.